testcases/GcovSetup.py

```diff
@@ -0,0 +1,185 @@
+        k_config_params = ["CONFIG_GCOV_KERNEL=y", "CONFIG_ARCH_HAS_GCOV_PROFILE_ALL=y", "CONFIG_GCOV_PROFILE_ALL=y", "CONFIG_GCOV_PROFILE_FTRACE=y"]
+        err_param = []
+        for param in k_config_params:
+            log.info("\n working on param  %s" %param)
+            unset_param = param
+            if param.split("=")[-1] == "y":
+                new_param = param[:-1] + "m"
+            else:
+                new_param = param[:-1] + "y"
+
+            out = int("".join(self.cv_HOST.host_run_command(f"grep -n ^{param} {conf_file} | wc -l")))
+            new_out = int("".join(self.cv_HOST.host_run_command(f"grep -n ^{new_param} {conf_file} | wc -l")))
+            if param.split("=")[-1] == "m":
+                new_unset_param = unset_param.split("=")[0]
+                unset_out = int("".join(self.cv_HOST.host_run_command(f"grep -n '{new_unset_param} is not set' {conf_file} | wc -l")))
+                if unset_out == 1:
+                    continue
+
+            if param.split("=")[-1] == "m":
+                param = f"#{param}"
+            if out == 1:
+                if param.split("=")[-1] == "m":
+                    log.info("param found but needs to unset....\n unsetting the param")
+                    self.cv_HOST.host_run_command(f"sed -i 's/^{unset_param}/{param}/g' {conf_file}")
+                else:
+                    log.info("%s parameter already available with same value, So skipping to change it" %param)
+                    continue
+            elif new_out == 1:
+                log.info("%s parameter exists with opposit valuue , changing it" %param)
+                self.cv_HOST.host_run_command(f"sed -i 's/^{new_param}/{param}/g' {conf_file}")
+            else:
+                log.info("%s parameter does not exists, so appending" %param)
+                self.cv_HOST.host_run_command(f"echo {param} >> {conf_file}")
+
+            out_check = int("".join(self.cv_HOST.host_run_command(f"grep -n ^{param} {conf_file} | wc -l")))
+            if out_check == 1:
+                log.info("%s  param added/changed successfully" %param)
+            else:
+                log.info("param failed to change {param}")
+                err_param.append(param)
+            log.info("\n\n\n")
+        if err_param:
+            self.fail("few param did not got updated: %s" %err_param)
+
+    def build_and_boot(self, config_path):
+        """
+        building and booting the newly installed kernel
+        """
+        try:
+            onlinecpus = int(self.cv_HOST.host_run_command("lscpu --online -e|wc -l")[-1])
+        except Exception:
+            onlinecpus = 20
+
+        self.cv_HOST.host_run_command(f"cd {config_path}")
+        self.cv_HOST.host_run_command("make olddefconfig")
+        try:
+            self.cv_HOST.host_run_command("make -j %s" %onlinecpus , timeout=self.host_cmd_timeout)
+            cmd = f"make -j modules_install && make install"
+            self.cv_HOST.host_run_command(cmd)
```

Comment:
can we check the return statement as well? as it capture op-test or python error

Reply:
so you want this to be 
`if self.cv_HOST.host_run_command(cmd):`

Reply:
yes as this help to capture command failed and early to fail without going to reboot cycle